From ea3935bd7c0f36c86c40e5785326698fa3336c6a Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@gentoo.org>
Date: Mon, 14 Nov 2016 22:48:42 -0500
Subject: [PATCH] iptunnel: fix building w/older linux headers

While linux-4.8+ headers work, older ones are missing includes.
Dropping netinet/ip.h uncovered that mess, so add linux/ip.h.

URL: https://bugs.gentoo.org/599542
Reported-by: Conrad Kostecki <ck@bl4ckb0x.de>
---
 iptunnel.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/iptunnel.c b/iptunnel.c
index 247ae10..d6e47e6 100644
--- a/iptunnel.c
+++ b/iptunnel.c
@@ -29,7 +29,12 @@
 #include <arpa/inet.h>
 #include <net/if.h>
 #include <net/if_arp.h>
+/* We only care about linux/if_tunnel.h.  Unfortunately, older Linux headers
+ * (pre linux-4.8) did not include all the proper headers leading to missing
+ * structs and types.
+ */
 #include <linux/types.h>
+#include <linux/ip.h>
 #include <linux/if_tunnel.h>
 
 #include "config.h"
-- 
2.9.0
